### PR TITLE
[Merged by Bors] - feat: more `Int.ediv` lemmas

### PR DESCRIPTION
**Pull request by eric-wieser**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by Ruben-VandeVelde — APPROVED** [review]
maintainer merge

File: Mathlib/Algebra/Order/Group/Unbundled/Int.lean [hunk 1]
```diff
@@ -129,6 +132,10 @@ protected theorem sign_eq_ediv_abs (a : ℤ) : sign a = a / |a| :=
   if az : a = 0 then by simp [az]
   else (Int.ediv_eq_of_eq_mul_left (mt abs_eq_zero.1 az) (sign_mul_abs _).symm).symm
 
+protected theorem sign_eq_abs_ediv (a : ℤ) : sign a = |a| / a :=
```
Comment (YaelDillies) [inline Mathlib/Algebra/Order/Group/Unbundled/Int.lean:135]: I think this one could be a good simp lemma backwards?

File: Mathlib/Data/Int/Lemmas.lean [hunk 2]
```diff
@@ -133,4 +133,16 @@ theorem div2_bit (b n) : div2 (bit b n) = n := by
 @[deprecated (since := "2024-04-02")] alias abs_coe_nat := abs_natCast
 @[deprecated (since := "2024-04-02")] alias coe_nat_nonpos_iff := natCast_nonpos_iff
 
+/-- Like `Int.ediv_emod_unique`, but permitting negative `b`. -/
+theorem ediv_emod_unique' {a b r q : Int} (h : b ≠ 0) :
+  a / b = q ∧ a % b = r ↔ r + b * q = a ∧ 0 ≤ r ∧ r < |b| := by
```
Comment (YaelDillies) [inline Mathlib/Data/Int/Lemmas.lean:138]:
```suggestion
    a / b = q ∧ a % b = r ↔ r + b * q = a ∧ 0 ≤ r ∧ r < |b| := by
```

File: Mathlib/Algebra/Order/Group/Unbundled/Int.lean (hunk 1 above)
Comment (eric-wieser) [reply Mathlib/Algebra/Order/Group/Unbundled/Int.lean:135]: Perhaps, but this matches the one above it, so that sounds like a follow-up

**Comment by kim-em** [comment]
bors merge

**Pull request closed without merging**